src/containers/flows/MakeABuy/MakeABuy.tsx

```diff
@@ -0,0 +1,494 @@
+/* eslint-disable react-hooks/exhaustive-deps */
+//- React Imports
+import React, { useState, useEffect } from 'react';
+
+//- Web3 Imports
+import { useWeb3React } from '@web3-react/core'; // Wallet data
+import { Web3Provider } from '@ethersproject/providers/lib/web3-provider'; // Wallet data
+
+//- Library Imports
+import { Metadata, Bid, Domain } from 'lib/types';
+import { useZnsContracts } from 'lib/contracts';
+import { getMetadata } from 'lib/metadata';
+import { BigNumber, ethers, providers } from 'ethers';
+import { ERC20 } from 'types';
+import { useBidProvider } from 'lib/providers/BidProvider';
+
+//- Component Imports
+import {
+	FutureButton,
+	Member,
+	NFTMedia,
+	Tooltip,
+	LoadingIndicator,
+} from 'components';
+
+//- Style Imports
+import styles from './MakeABuy.module.scss';
+import useCurrency from 'lib/hooks/useCurrency';
+import { useZnsSdk } from 'lib/providers/ZnsSdkProvider';
+
+interface MakeABuyProps {
+	domain: Domain;
+	onBuy: () => void;
+}
+
+const MakeABuy: React.FC<MakeABuyProps> = ({ domain, onBuy }) => {
+	//- Bid hooks
+	const { getBidsForDomain } = useBidProvider();
+
+	// Wild to usd
+	const { wildPriceUsd } = useCurrency();
+
+	///////////
+	// State //
+	///////////
+
+	const [isBuyNowFinishedSuccesfully, setBuyNowFinishedSuccesfully] =
+		useState(false);
+
+	// todo add shopWalletInteraction if needed
+	// const [ShowWalletInteraction, setShowWalletInteraction] = useState(false);
+	const [ShowProccesing, setShowProccesing] = useState(false);
+
+	const [buyNowPrice, setBuyNowPrice] = useState<BigNumber | undefined>();
+	const [isActionDeclined, setActionDeclined] = useState(false);
+	const [buyNowPriceUsd, setBuyNowPriceUsd] = useState<number | undefined>();
+	const [currentHighestBid, setCurrentHighestBid] = useState<Bid | undefined>();
+	const [currentHighestBidUsd, setCurrentHighestBidUsd] = useState<
+		number | undefined
+	>();
+	const [wildBalance, setWildBalance] = useState<number | undefined>();
+	const [domainMetadata, setDomainMetadata] = useState<Metadata | undefined>();
+
+	// Loading States
+	const [isMetamaskWaiting, setIsMetamaskWaiting] = useState(false);
+
+	//- Web3 Wallet Data
+	const walletContext = useWeb3React<Web3Provider>();
+	const { account, library } = walletContext;
+
+	const znsContracts = useZnsContracts()!;
+	const wildContract: ERC20 = znsContracts.wildToken;
+
+	const [isBuyValid, setBuyValid] = useState(false);
+
+	const sdk = useZnsSdk();
+	const domainId = domain.id;
+
+	const provider = library && new providers.Web3Provider(library.provider);
+	const signer = provider && provider.getSigner(account!);
```

Comment:
Please updated this to	
```const { account, active, library, chainId } = useWeb3React<Web3Provider>(); await library.getSigner().```

Reply:
Actually i only need signer for the zAuction methods and have moved it to the hook as well